File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int n) {
        int[] answer = {};
        ArrayList<Integer> myArr = new ArrayList<>();
        while (n != 1) {
            myArr.add(n);
            n = n % 2 == 0 ? n / 2 : 3 * n + 1;
        }
        myArr.add(1);
        
        answer = myArr.stream().mapToInt(i -> i).toArray();
        return answer;
    }
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n):
    answer = []
    answer.append(n)
    
    while (n != 1 and n != 0):
        if n % 2 == 0:
            # x가 짝수일 때는 2로 나누고, 
            n = n / 2
            answer.append(n)
        elif n % 2 == 1:
            # x가 홀수일 때는 3 * x + 1로 바꾸는 계산
            n = n * 3 + 1
            answer.append(n)
            
    return answer